feat: add settings and detail page to apps

klmhsb42 opened the pull request:
Features:

- Filter Apps by Categories
- Sort Apps (by A-Z or Categories)
- Toggle Grid or List View
- Detail page of apps (Load Screenshots from external,  Load Readme from external, Click Links to Project or App, Share an App, Embed Link of your App in your website)
- Manage all App Metadata in apps.json

Todos:
- some bug fixes before merge (draft status)

Screenshots:

![Bildschirmfoto vom 2024-10-26 11-55-42](https://github.com/user-attachments/assets/b91be9fd-ff70-44f7-8e2f-5af1fa8f00cd)
![Bildschirmfoto vom 2024-10-26 11-57-25](https://github.com/user-attachments/assets/4e2af4e8-cbf6-4094-90d4-3ffef340a50b)
![Bildschirmfoto vom 2024-10-26 11-58-34](https://github.com/user-attachments/assets/b1075ceb-c588-4fac-abca-1e021772e65f)
![Bildschirmfoto vom 2024-10-26 11-58-42-obfuscated](https://github.com/user-attachments/assets/4f9e11c1-11d7-4e59-b46a-17a647bb9600)
![Bildschirmfoto vom 2024-10-26 11-58-54](https://github.com/user-attachments/assets/033560ab-53e9-473d-bde4-faff03c80f80)

cyberb commented:
Also I am mostly using mobile so you need to make sure it does not break it

klmhsb42 commented:
Thanks for feedback. Sounds like you are ok with the feature ideas in general? So I will implement your comments and make bug fixes and get back to you for final approval.

cyberb commented:
> Sounds like you are ok with the feature ideas in general?

Yes, it just never made to the top of the priority.
I would like to point out that one of the reasons I was not doing this on this web site as I always wanted to do it properly on its own store web where all the details would come from the app package itself (like on google play). There is already a store api it just does not have its own web.
Anyway if and when I get to the store web we can use what your are doing right so it will not be wasted.

cyberb commented:
Also grid switch is a bit odd comparing to the other two controls.

cyberb commented:
In general it is difficult to actually create something better and maintainable without using a proper web framework. On device for example (https://github.com/syncloud/platform/tree/master/www) I am using Vue.js v3 with element-plus component library (https://element-plus.org/en-US/component/overview.html) which gives a lot of power. I am not saying it has to be used right now but it would give you an easy access to a set of components, but it can be done later when moving to store web.

klmhsb42 commented:
Hi @cyberb , I pushed a new version. I think it looks much better now and I also tested it with a smartphone. It's not perfect and certainly needs improvements but I think it's good enough for a first release. I also added a search function. Can you check it out?

I totally agree with you. Maybe we can proceed as follows:

1) First take this version
2) Replace the app JSON in https://github.com/syncloud/release/blob/master/index-v2 with the new JSON and then completely remove it from Syncloud.org and make some changes in Syncloud.org repo to access it to have only single source of truth
3) Then add the new app store functions in the platform app and also rewrite them with vue.js functions
4) Then transfer the vue.js functions to Syncloud.org

cyberb commented:
![image](https://github.com/user-attachments/assets/961dc4f1-65f1-4031-99fe-43b61f6308d7)

I would say these days you do not need to say "Filter Category" or "Sort" it is all intuitive and will look better without them

cyberb commented:
![image](https://github.com/user-attachments/assets/8eeb0c6d-659e-4e7b-8849-45a9bb63b6e1)

I think this one is good on the wide view.

![image](https://github.com/user-attachments/assets/381c61a0-615a-4aea-8502-8b029a8e13a8)

But on mobile view it is not so good, can we make it look the same as on wide view (on the same line)?

cyberb commented:
![image](https://github.com/user-attachments/assets/69e7fcaa-b075-4fe6-8ee6-371c91952aa4)

I think these days sites try to be simpler and have a single scroll (the main one) by not limiting vertical size of any component, this would have less confusion of what to scroll by placing the mouse or a finger in different areas of the screen. 
In simple implementation that usually means filter, scroll down, then scroll up to see filters again, but that can be fixed by freezing the filter controls at the top (again not very convenient thing as it takes part of your screen which is very annoying).

So I would prefer filters at the top of the page which are moved away with the page as you scroll, simple and very practical (in my mind)

cyberb commented:
![image](https://github.com/user-attachments/assets/181832a8-ddbe-41c6-98aa-e3fcc68f8ca7)

Colors are different, I think the top one is the official color we use, let's stick to that.
There is nothing worse than inconsistent colors on a website (I am not saying I have not done this mistake myself, but it is still a mistake :) )

klmhsb42 commented:
> I would say these days you do not need to say "Filter Category" or "Sort" it is all intuitive and will look better without them. Also all in one horizontal line would be better on this wide view.

Removed and in one line.

> I think this one is good on the wide view.
> 
> But on mobile view it is not so good, can we make it look the same as on wide view (on the same line)?

Also in mobile on same line now.

> I think these days sites try to be simpler and have a single scroll (the main one) by not limiting vertical size of any component, this would have less confusion of what to scroll by placing the mouse or a finger in different areas of the screen. In simple implementation that usually means filter, scroll down, then scroll up to see filters again, but that can be fixed by freezing the filter controls at the top (again not very convenient thing as it takes part of your screen which is very annoying).
> 
> So I would prefer filters at the top of the page which are moved away with the page as you scroll, simple and very practical (in my mind)

Done. Just one scroll bar now.

> Colors are different, I think the top one is the official color we use, let's stick to that. There is nothing worse than inconsistent colors on a website (I am not saying I have not done this mistake myself, but it is still a mistake :) )

Color is now corrected for buttons.

cyberb commented:
Hi, all good!
Can you fix this last minor thing, alignment on left side is not right?
![Screenshot from 2024-11-29 08-37-35](https://github.com/user-attachments/assets/ad17f9cf-e657-4928-9d78-09186d803fae)

klmhsb42 commented:
Hi @cyberb , the details usually take the most time. It should now be pixel accurate. I hope it works.
![Bildschirmfoto vom 2024-12-01 11-53-31](https://github.com/user-attachments/assets/91106884-ba79-483a-aaae-01d6c119412e)

cyberb commented:
ok design is good, but functionality is a bit strange, category and name filter do not play well together.
[Kooha-2024-12-02-21-04-23.webm](https://github.com/user-attachments/assets/87a39699-c590-498b-b575-8b7141ef676c)

klmhsb42 commented:
@cyberb should work better now

cyberb commented:
Sorry forgot about this PR, everything seems to work.
Couple things about the code:
1. Browser throws an error
![image](https://github.com/user-attachments/assets/ad10ba61-ee70-448a-a95d-a3f28ce653be)
2. Can you format app.js/app-details.js files please as indentation is very inconsistent?
